Cargo.toml

```diff
@@ -22,6 +26,10 @@ dusk-plonk = { git = "https://github.com/maticnetwork/plonk.git", branch = "v0.8
 frame-system = { path = "pallets/system" }
 frame-executive = { path = "pallets/executive" }
 frame-system-rpc-runtime-api = { path = "pallets/system/rpc/runtime-api" }
+syn = { git = "https://github.com/dtolnay/syn.git", tag = "1.0.96" }
```

Comment:
Is coupling to github repo and specific tag needed for some reason? (same question is for other deps)

Reply:
Any kinds of tag/version pinning was to avoid conflicts between _external_ crates that are on different versions than those used in substrate by default. Removing the patch lines will result in inability to compile

Reply:
We could specify exact versions in crates which fails to compile, to avoid hard coupling to specific source (and allow updates from crates.io), if I understand this correctly.

e.g. commenting all four lines and adding 
```
# Set to exact versions due to broken compatibility
syn = "=1.0.96"
parity-db = "=0.3.12"
```
to primitives/nomad/merkle/Cargo.toml fixes compile errors. But I don't know if and how this affects next PR :)

Reply:
Lets address this in next PR if possible.